DatadogInternal/Sources/NetworkInstrumentation/TraceID.swift

```diff
@@ -38,17 +107,41 @@ extension TraceID {
     public init?(_ string: String, representation: Representation = .decimal) {
         switch representation {
         case .decimal:
-            guard let rawValue = UInt64(string) else {
+            guard let idLo = UInt64(string) else {
+                return nil
+            }
+
+            self.init(idLo: idLo)
+        case .hexadecimal16Chars:
+            guard let idLo = UInt64(string, radix: 16) else {
                 return nil
             }
 
-            self.init(rawValue: rawValue)
-        case .hexadecimal, .hexadecimal16Chars, .hexadecimal32Chars:
-            guard let rawValue = UInt64(string, radix: 16) else {
+            self.init(idLo: idLo)
+        case .hexadecimal32Chars:
+            guard let idLo = UInt64(string, radix: 16), let idHi = UInt64(string.prefix(16), radix: 16) else {
                 return nil
             }
 
-            self.init(rawValue: rawValue)
+            self.init(rawValue: (idHi, idLo))
```

Comment:
suggestion/ These two seem to have no unit tests coverage (in `DatadogInternalTests`). Overall score for `TraceID` seems low given that this is critical piece. Could we improve it?

<img width="759" alt="Screenshot 2024-04-09 at 10 28 51" src="https://github.com/DataDog/dd-sdk-ios/assets/2358722/b0b0b782-df9b-4757-aefe-f332dedbcc39">

Reply:
fair call, I added bunch of cases now.